### PR TITLE
Added arguments for altering color palettes of CrunchBoxes

### PR DESCRIPTION
**Pull request by jonkeane**

### CODE REVIEW

File: R/crunchbox.R [hunk 1]
```diff
@@ -5,11 +5,16 @@
...
+#' @param brand_colors an optional list with colors (in hex format) named 'primary', 'secondary', and 'message'
+#' @param static_colors an optional list of static colors (in hex format) to use for categoricals (taken in order)
+#' @param category_color_lookup an optional list of category names to colors (in hex format) to use for that category (for fine grained tuning of category colors.)
 #' @param ... additional metadata for the box, such as "title", "header", etc.
 #' @return The URL to the newly created box.
 #' @seealso \code{\link{preCrunchBoxCheck}} to provide guidance on what you're including in the CrunchBox
 #' @export
-crunchBox <- function (dataset, filters=crunch::filters(dataset), ...) {
+crunchBox <- function (dataset, filters=crunch::filters(dataset),
+                       brand_colors, static_colors,
```
Comment (GShotwell) [inline R/crunchbox.R]: I think it would be helpful to have an example of changing the colour scheme. You can probably just copy the ones from the tests.

File: R/crunchbox.R [hunk 2]
```diff
@@ -41,6 +46,28 @@ crunchBox <- function (dataset, filters=crunch::filters(dataset), ...) {
     ## Add "where" after so that it no-ops if variablesFilter returns NULL (i.e. no filter)
     payload$where <- variablesFilter(dataset)
 
+    ## Add colors if they exist to the payload
+    if (!missing(brand_colors)) {
+        if (!is.list(brand_colors) ||
+            any(!names(brand_colors) %in% c("primary", "secondary", "message"))) {
+            halt(sQuote("brand_colors"), " must be a named list with only ",
+                 serialPaste(dQuote(c("primary", "secondary", "message")), collapse = "or"))
```
Comment (GShotwell) [inline R/crunchbox.R]: Is there any way we can pass the colour scheme without a named list? For instance a character vector of length three which represented the three colours? I'm thinking of `ggplot2::scale_colour_manual()` as an example. Named lists can be a bit tricky for some users. Also a character vector would eliminate problems with  people naming the list incorrectly.

File: R/crunchbox.R [hunk 3]
```diff
@@ -41,6 +46,28 @@ crunchBox <- function (dataset, filters=crunch::filters(dataset), ...) {
...
+        }
+        payload$display_settings$palette$brand_colors <- brand_colors
+    }
+    if (!missing(static_colors)) {
+        if (!is.list(static_colors)) {
+            halt(sQuote("static_colors"), " must be a list of characters")
```
Comment (GShotwell) [inline R/crunchbox.R]: Would a character vector be okay here? If so I think that's probably easier than a list. Also the error specifies that it be a list of character strings, but I don't think that's actually being checked.

File: R/crunchbox.R [hunk 4]
```diff
@@ -41,6 +46,28 @@ crunchBox <- function (dataset, filters=crunch::filters(dataset), ...) {
...
+        }
+        payload$display_settings$palette$static_colors <- static_colors
+    }
+    if (!missing(category_color_lookup)) {
+        if (!is.list(category_color_lookup) || is.null(names(category_color_lookup))) {
```
Comment (GShotwell) [inline R/crunchbox.R]: Do we want to try to check that the categories supplied in the list match categories in the data? Or if there are mismatches on either side?

File: tests/testthat/test-crunchbox.R [hunk 5]
```diff
@@ -111,12 +111,57 @@ with_mock_crunch({
...
         expect_error(crunchBox(), "'dataset' must be a CrunchDataset")
         expect_error(crunchBox(4), "'dataset' must be a CrunchDataset")
         expect_error(crunchBox(ds, 4),
             "'filters' should be a FilterCatalog or NULL")
-
+        expect_error(crunchBox(ds, brand_colors = "a"),
+                     paste0(
+                         sQuote("brand_colors"), " must be a named list with only ",
+                         serialPaste(dQuote(c("primary", "secondary", "message")),
+                                     collapse = "or")))
```
Comment (GShotwell) [inline tests/testthat/test-crunchbox.R]: You might want to add some tests for non-characters being passed as arguments like number, NA, NULL

File: R/crunchbox.R (hunk 1 above)
Comment (nealrichardson) [reply R/crunchbox.R]: 👍

File: R/crunchbox.R (hunk 4 above)
Comment (nealrichardson) [reply R/crunchbox.R]: I think that might be too expensive (but perhaps something to add to the `preCrunchBoxCheck`?)

File: tests/testthat/test-crunchbox.R (hunk 5 above)
Comment (nealrichardson) [reply tests/testthat/test-crunchbox.R]: And that they're actually valid colors

File: R/crunchbox.R (hunk 2 above)
Comment (nealrichardson) [reply R/crunchbox.R]: 👍

File: R/crunchbox.R (hunk 4 above)
Comment (jonkeane) [reply R/crunchbox.R]: I've added this as a todo in precrunchbox check for now.

File: tests/testthat/test-crunchbox.R (hunk 5 above)
Comment (jonkeane) [reply tests/testthat/test-crunchbox.R]: Done and done.
